File: rei-do-chopp-site/src/app/orders/models/responses/order-product-history.response.ts
import { ProductResponse } from "../../../products/models/responses/product.response";

export class OrderProductHistoryResponse {
  Id: number;
  OrderId: number;
  ProductId: number;
  Product: ProductResponse;
  Quantity: number;
  UnitPriceCharged: number;
  ExpectedUnitPrice: number;
  TotalPriceCharged: number;
  ExpectedTotalPrice: number;
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/OrdersProducts/Responses/OrderProductHistoryResponse.cs
using ReiDoChopp.DataTransfer.Products.Response;

namespace ReiDoChopp.DataTransfer.OrdersProducts.Response
{
    public class OrderProductHistoryResponse
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public ProductResponse Product { get; set; }
        public int Quantity { get; set; }
        public double UnitPriceCharged { get; set; }
        public double ExpectedUnitPrice { get; protected set; }
        public double TotalPriceCharged { get; protected set; }
        public double ExpectedTotalPrice { get; protected set; }
    }
}
